eslint.config.mjs

```diff
@@ -7,6 +7,24 @@ import barrel from 'eslint-plugin-barrel-files'
 import eslint from '@eslint/js'
 import tseslint from 'typescript-eslint'
 import storybook from 'eslint-plugin-storybook'
+import { readdirSync } from 'fs'
+
+const getDirectories = source =>
+  readdirSync(source, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name)
+
+// TODO: Use Nx API to retrieve project directories
+const appDirs = getDirectories(`${import.meta.dirname}/apps`).map(
+  app => `apps/${app}`,
+)
+const portalDirs = getDirectories(`${import.meta.dirname}/apps/portals`).map(
+  app => `apps/portals/${app}`,
+)
+const packageDirs = getDirectories(`${import.meta.dirname}/packages`).map(
+  pkg => `packages/${pkg}`,
+)
+const allProjectDirs = [...appDirs, ...portalDirs, ...packageDirs]
```

Comment:
Nx has a really friendly command-line API but an unintuitive JavaScript API for building things that are not plugins. Using Nx would be the right way to do this but I couldn't figure it out.